### DISCUSSION TITLE
How to register material-ui Slider component?

### DISCUSSION THREAD

**Discussion by ertw** [react-hook-form/react-hook-form#2712] [Q&A]
Hi,

I'm wondering if the controller ... as pattern is still available?

This example similar to the documentation works in react-hook-form 5.5.1, but not in 6.5.3:
```jsx
<Controller
  name="MUI_Slider"
  control={control}
  defaultValue={[0, 10]}
  onChange={([, value]) => value}
  as={<Slider valueLabelDisplay="auto" max={10} step={1} />}
 />
```

Here is the working example: https://codesandbox.io/s/react-hook-form-controlled-slider-8y06j?file=/src/index.js

If I update the react-hook-form dependency to the current version (6.5.3), it doesn't work and I see console errors from both react-hook-form and mui:

```
📋 Controller `defaultValue` or useForm `defaultValues` is missing. https://react-hook-form.com/api#Controller 
    in Controller (at src/​index.js:42)
    in App (at src/​index.js:59)
```

```
index.js:27 Material-UI: A component is changing the controlled value state of Slider to be uncontrolled.
Elements should not switch from uncontrolled to controlled (or vice versa).
Decide between using a controlled or uncontrolled Slider element for the lifetime of the component.
The nature of the state is determined during the first render, it's considered controlled if the value is not `undefined`.
More info: https://fb.me/react-controlled-components 
```

Any ideas on how I can use a mui Slider component?

**Comment by bluebill1049** [comment]
use `render` prop. take a look the official doc, i think there is an example.

**Reply by goldpot** [reply]
@ render props does not work when we want to have our own function for onChange and not use props.onChange - something like this - 

<Grid item xs>
                                        <Controller control={methods.control} name={option.label}
                                            defaultValue={option.default_value}
                                            render={(props) => (
                                                <Slider
                                                    {...props}
                                                    onChange={(_, value) => {
                                                        props.onChange(value); // here want to use onSliderChange like in commented code below

                                                    }}
                                                    valueLabelDisplay="auto"
                                                    max={option.choices[1]}
                                                    min={option.choices[0]}
                                                    step={1}
                                                />

                                            )}
                                        />
                                        {/* max={option.choices[1]}
                                             min={option.choices[0]}
                                             defaultValue={option.default_value}
                                            value={state.minMapqToCount}
                                             onChange={(e) => onSliderChange(option.name, e)}  */}

                                        {/* // <Slider
                                        //     name={option.name}
                                        //     max={option.choices[1]}
                                        //     min={option.choices[0]}
                                        //     // defaultValue={option.default_value}
                                        //     value={state.minMapqToCount}
                                        //     onChange={(e) => onSliderChange(option.name, e)}
                                        // /> */}
                                    </Grid>

Here I want to use onSliderChange function for my onChange - and it does not work ... I have been trying this for a while now and it does not look like react hook supports this.. can you please help me out here? Thanks

**Reply by bluebill1049** [reply]
1. there is an example of CSB in the doc: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r
2. you can always have your own onChange={(v) => props.onChange(v) && doYourOwnThing}
